File: auth-service/src/main/java/com/smartops/auth/service/ProfileService.java
package com.smartops.auth.service;



import com.smartops.auth.dto.ChangePasswordRequest;
import com.smartops.auth.dto.ProfileRequest;
import com.smartops.auth.dto.ProfileResponse;
import com.smartops.auth.model.User;
import com.smartops.auth.repo.UserRepo;
import lombok.RequiredArgsConstructor;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class ProfileService {

    private final UserRepo userRepository;

    private final PasswordEncoder passwordEncoder;

    public ProfileResponse getProfile(Long userId) {

        User user = userRepository.findById(userId)
                .orElseThrow(() ->
                        new RuntimeException("User not found"));

        return mapToResponse(user);
    }

    public ProfileResponse updateProfile(
            Long userId,
            ProfileRequest request
    ) {

        User user = userRepository.findById(userId)
                .orElseThrow(() ->
                        new RuntimeException("User not found"));

        user.setName(request.getName());
        user.setEmail(request.getEmail());

        userRepository.save(user);

        return mapToResponse(user);
    }

    public void changePassword(
            Long userId,
            ChangePasswordRequest request
    ) {

        User user = userRepository.findById(userId)
                .orElseThrow(() ->
                        new RuntimeException("User not found"));

        if (!passwordEncoder.matches(
                request.getCurrentPassword(),
                user.getPassword()
        )) {
            throw new RuntimeException(
                    "Current password is incorrect"
            );
        }

        user.setPassword(
                passwordEncoder.encode(
                        request.getNewPassword()
                )
        );

        userRepository.save(user);
    }

    private ProfileResponse mapToResponse(User user) {

        return ProfileResponse.builder()
                .id(user.getId())
                .name(user.getName())
                .email(user.getEmail())
                .role(user.getRole().name())
                .build();
    }
}

File: smartops-dashboard/services/profile-service.ts
import api from '@/service/api';

export interface Profile {
  id: number;
  name: string;
  email: string;
  role: string;
}

export interface UpdateProfileRequest {
  name: string;
  email: string;
}

export interface ChangePasswordRequest {
  currentPassword: string;
  newPassword: string;
}

export const profileService = {
  async getProfile(): Promise<Profile> {
    const response = await api.get('/api/auth/profile');
    return response.data;
  },

  async updateProfile(
    data: UpdateProfileRequest
  ): Promise<Profile> {
    const response = await api.put(
      '/api/auth/profile',
      data
    );

    return response.data;
  },

  async changePassword(
    data: ChangePasswordRequest
  ): Promise<string> {
    const response = await api.put(
      '/api/auth/profile/change-password',
      data
    );

    return response.data;
  },
};